src/java.base/share/classes/java/nio/channels/spi/SelectorProvider.java

```diff
@@ -95,12 +95,10 @@ private static SelectorProvider loadProviderFromProperty() {
             try {
                 Class<?> clazz = Class.forName(cn, true, ClassLoader.getSystemClassLoader());
                 return (SelectorProvider) clazz.getConstructor().newInstance();
-            } catch (ClassNotFoundException |
-                    NoSuchMethodException |
-                    IllegalAccessException |
-                    InvocationTargetException |
-                    InstantiationException x) {
-                throw new ServiceConfigurationError(null, x);
+            } catch (Throwable t) {
```

Comment:
This fix should have a regression test, based on the provided reproducer.
If it can't be implemented, then bug should have 'noreg-' label.